test/e2e/util.go

```diff
@@ -161,6 +161,10 @@ type TestContextType struct {
 	GatherMetricsAfterTest            bool
 	// Currently supported values are 'hr' for human-readable and 'json'. It's a comma separated list.
 	OutputPrintType string
+	// CreateTestingNS is responsible for creating namespace used for executing e2e tests.
+	// It accepts namespace base name, which will be prepended with e2e prefix, kube client
+	// and labels to be applied to a namespace.
+	CreateTestingNS func(baseName string, c *client.Client, labels map[string]string) (*api.Namespace, error)
```

Comment:
Original goal is to enable the conformance tests to run against
clusters that require elevated security permissions to do unsafe
things (OpenShift is a conforming k8s cluster but does not allow
hostPath or root container access to regular users).  So we grant
elevated privileges when a namespace is created - this wrapper allows
us to ensure those are in place prior to test startup.  There are
other bits too - being able to assign default permissions depending on
the test type, and ensuring that other supporting resources are in
place like quota and limit ranges.  So generally this allows the
conformance suite to run correctly and be extended as implementations
provide their own security models.